src/Actor.cpp

```diff
@@ -520,17 +520,15 @@ void Actor::PreDraw() // calculate actor properties
 		const float rupathrdown_plus_atf= rupath_plus_rdown + m_effect_hold_at_full;
 		if(fTimeIntoEffect < m_effect_ramp_to_half)
 		{
-			fPercentThroughEffect = SCALE(fTimeIntoEffect,
-				0, m_effect_ramp_to_half, 0.0f, 0.5f);
+			fPercentThroughEffect = (((fTimeIntoEffect)-(0)) * ((0.5f) - (0.0f)) / ((m_effect_ramp_to_half)-(0)) + (0.0f));
```

Comment:
Without looking at every occurrence, there are now a bunch of extra parenthesis that should be removed.

Reply:
Actually I think just having an inline function the same as scale is probably more readable than doing trying to parse this,

Reply:
> Actually I think just having an inline function the same as scale is probably more readable than doing trying to parse this,

No template function could satisfy all cases of SCALE() where it's being called. this is because the macro does not do type checking. I am worried about adding new `static_cast`s to places and potentially changing behavior. Would appreciate any opinions here -  do you think it's a concern? Would it be better to clean up all the inlined functions to remove unnecessary parenthesis and parts of the function?

Reply:
The concern is mostly that when I see something like:
`(((fTimeIntoEffect)-(0)) * ((0.5f) - (0.0f)) / ((m_effect_ramp_to_half)-(0)) + (0.0f))`
I have to do a *lot* of mental gymnastics to figure out what it's doing here.

You don't have to template the scale function here. You can define your multiple like `Scale(int, int, int, int)` or `Scale(float, float, float, float)` and another for doubles etc and the compiler can figure out which one it should use.